sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -610,8 +610,9 @@ pi_result _pi_queue::executeCommandList(ze_command_list_handle_t ZeCommandList,
                                         bool IsBlocking,
                                         bool OKToBatchCommand) {
   if (OKToBatchCommand && this->isBatchingAllowed()) {
-    assert(this->ZeOpenCommandList == nullptr ||
-           this->ZeOpenCommandList == ZeCommandList);
+    if (this->ZeOpenCommandList != nullptr &&
+        this->ZeOpenCommandList != ZeCommandList)
+      return PI_INVALID_VALUE;
```

Comment:
Things like this, which aren't given by the user (SYCL RT) should not be reported back as errors. I believe the right strategy for these is to "die" instead.

Reply:
I might have missed few such cases. I am double checking.